### PR TITLE
7234 migrate edit channel header modal

### PR DESCRIPTION
**Pull request by Zapix**
#### Summary
Updated EditChannelHeaderModalComponent with requirements. 
Connected them with redux store.
#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/7234

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes

### CODE REVIEW

File: components/edit_channel_header_modal/edit_channel_header_modal.jsx [hunk 1]
```diff
@@ -25,43 +21,49 @@ const holders = defineMessages({
...
+class EditChannelHeaderModal extends React.PureComponent {
+    static propTypes = {
+        intl: intlShape.isRequired,
+        onHide: PropTypes.func.isRequired,
+        channel: PropTypes.object.isRequired,
+        ctrlSend: PropTypes.bool.isRequired,
+        serverError: PropTypes.object,
+        requestStatus: PropTypes.string.isRequired,
+        onPatchChannel: PropTypes.func.isRequired
+    }
```
Comment (saturninoabril) [inline components/edit_channel_header_modal/edit_channel_header_modal.jsx]: Please add short description/comment about each prop.

File: components/edit_channel_header_modal/edit_channel_header_modal.jsx [hunk 2]
```diff
@@ -25,43 +21,49 @@ const holders = defineMessages({
...
         super(props);
 
         this.handleChange = this.handleChange.bind(this);
         this.handleSave = this.handleSave.bind(this);
         this.handleKeyDown = this.handleKeyDown.bind(this);
         this.handleKeyPress = this.handleKeyPress.bind(this);
-        this.onShow = this.onShow.bind(this);
         this.onHide = this.onHide.bind(this);
-        this.handlePostError = this.handlePostError.bind(this);
         this.focusTextbox = this.focusTextbox.bind(this);
```
Comment (saturninoabril) [inline components/edit_channel_header_modal/edit_channel_header_modal.jsx]: Should remove all function binding. (Prefer `fn = () => {}`, see requested change below.)

File: components/edit_channel_header_modal/edit_channel_header_modal.jsx [hunk 3]
```diff
@@ -25,43 +21,49 @@ const holders = defineMessages({
...
+        if (requestStatus !== nextRequestStatus && nextRequestStatus === RequestStatus.FAILURE) {
+            this.setState({showError: true});
+        } else if (requestStatus !== nextRequestStatus && nextRequestStatus === RequestStatus.SUCCESS) {
+            this.onHide();
+        } else {
+            this.setState({showError: false});
+        }
     }
 
     handleChange(e) {
```
Comment (saturninoabril) [inline components/edit_channel_header_modal/edit_channel_header_modal.jsx]: Should change to `handleChange = (e) => {`

File: components/edit_channel_header_modal/edit_channel_header_modal.jsx [hunk 4]
```diff
@@ -70,33 +72,10 @@ class EditChannelHeaderModal extends React.Component {
...
-
-    onShow() {
-        this.submitted = false;
+    handleSave = () => {
+        const {channel, onPatchChannel: patchChannel} = this.props;
+        const {header} = this.state;
+        patchChannel(channel.id, {header});
     }
 
     onHide() {
```
Comment (saturninoabril) [inline components/edit_channel_header_modal/edit_channel_header_modal.jsx]: Should change to `onHide = () => {`

File: components/edit_channel_header_modal/edit_channel_header_modal.jsx [hunk 5]
```diff
@@ -110,13 +89,15 @@ class EditChannelHeaderModal extends React.Component {
     }
 
     handleKeyDown(e) {
```
Comment (saturninoabril) [inline components/edit_channel_header_modal/edit_channel_header_modal.jsx]: Should change to `handleKeyDown = (e) => {`

File: components/edit_channel_header_modal/edit_channel_header_modal.jsx [hunk 6]
```diff
@@ -110,13 +89,15 @@ class EditChannelHeaderModal extends React.Component {
...
-        if (this.state.ctrlSend && e.keyCode === KeyCodes.ENTER && e.ctrlKey === true) {
+        const {ctrlSend} = this.props;
+        if (ctrlSend && e.keyCode === KeyCodes.ENTER && e.ctrlKey === true) {
             this.handleKeyPress(e);
         }
     }
 
     handleKeyPress(e) {
```
Comment (saturninoabril) [inline components/edit_channel_header_modal/edit_channel_header_modal.jsx]: Should change to `handleKeyPress = (e) => {`

File: components/edit_channel_header_modal/index.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        ...submitInfo,
+    }),
+);
+
+export default connect(
+    mapStateToProps,
+    {
+        onPatchChannel: patchChannel
+    }
+)(EditChannelHeaderModal);
```
Comment (saturninoabril) [inline components/edit_channel_header_modal/index.js]: Should have implement mapDispatchToProps like:
```
function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            patchChannel
        }, dispatch)
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(EditChannelHeaderModal);
```

File: components/edit_channel_header_modal/index.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+import {connect} from 'react-redux';
+import {createSelector} from 'reselect';
+
```
Comment (saturninoabril) [inline components/edit_channel_header_modal/index.js]: add `import {bindActionCreators} from 'redux';`

File: tests/components/edit_channel_header_modal.test.jsx [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+                onPatchChannel={patchChannel}
+            />
+        ).dive({disableLifecycleMethods: true});
+
+        wrapper.find('.save-button').simulate('click');
+
+        expect(patchChannel.mock.calls.length).toBe(1);
+        expect(patchChannel.mock.calls[0][0]).toBe('fake-id', {header: 'Fake Channel'});
+    });
+});
```
Comment (saturninoabril) [inline tests/components/edit_channel_header_modal.test.jsx]: Test coverage for this component is a bit low. If possible, please add more tests to bring it up. Please see this [test](https://github.com/saturninoabril/mattermost-webapp/blob/3272562cf4cee3092ce6b5ede4fb4035704d1fd8/tests/components/post_view/post_info/post_info.test.jsx) if similar can be applied here.
screen shot 2017-10-07 at 1 26 24 am
screen shot 2017-10-07 at 1 38 54 am

File: components/edit_channel_header_modal/edit_channel_header_modal.jsx [hunk 10]
```diff
@@ -25,43 +21,49 @@ const holders = defineMessages({
...
-class EditChannelHeaderModal extends React.Component {
+class EditChannelHeaderModal extends React.PureComponent {
+    static propTypes = {
+        intl: intlShape.isRequired,
+        onHide: PropTypes.func.isRequired,
+        channel: PropTypes.object.isRequired,
+        ctrlSend: PropTypes.bool.isRequired,
+        serverError: PropTypes.object,
+        requestStatus: PropTypes.string.isRequired,
+        onPatchChannel: PropTypes.func.isRequired
```
Comment (saturninoabril) [inline components/edit_channel_header_modal/edit_channel_header_modal.jsx]: Might need to change like:
```
actions: PropTypes.shape({
    patchChannel: PropTypes.func.isRequired
}).isRequired
```

**Comment by Zapix** [comment]
@saturninoabril Updated with your requested changes. Also increased test coverage for component

 
coverage

**Comment by saturninoabril** [comment]
Awesome, thanks! It's great!

**Comment by saturninoabril** [comment]
I think it's failing to eslint when you run `make check-style`.

**Comment by Zapix** [comment]
@saturninoabril thank you. will add pre-commit hook to avoid style/test errors :)

**Pull request merged**